src/ptrmask-integration.adoc

```diff
@@ -0,0 +1,16 @@
+[#section_pointer_masking_integration]
+== Integrating {cheri_base_ext_name} and {cheri_default_ext_name} with Pointer Masking
+
+The pointer masking extensions Smmpm, Smnpm, SSnpm, Sspm and Supm are compatible with {cheri_default_ext_name}.
+
+In {cheri_int_mode_name} all data addresses are integer pointers, which
+are interpreted as being XLEN-wide, and may be subject to pointer masking.
+All data accesses are checked against <<ddc>> which is unaffected by
+pointer masking. Therefore no capability bounds encoding is affected.
+
+In {cheri_cap_mode_name} all pointers are capabilities. If pointer masking is enabled
```

Comment:
This is the way that Morello did it, but they faced some unintended consequences as a result. Because of how sign extension would change bounds decode, modifying a capability address (completely in bounds) can result in dramatically different bounds. This was even a bug for a while as it beat the fast representability check. They ended up having to special case changing the 56th bit of address, if the capability had a specific range of E's, to always untag when modified even though the capability was never taken out of bounds.

I think we should probably learn from this lesson that whilst the address should eventually be sign extended for the purpose of generating a vaddr for the next stage of memory fetch, sign extension should likely not be a part of bounds calculation / bounds check so that in-bounds capabilities are always representable.

I would suggest for the purpose of bounds decode / bounds check high bits just be masked (not extended)

Reply:
Agreed we need to fix this for a proper integration, but for now just doing the bare minimum makes the most sense.